From 7e3b36f4a6bfd2f8c166e4b6a972bac9a455e142 Mon Sep 17 00:00:00 2001
From: Jean Trivelly <jean.trivelly@intel.com>
Date: Mon, 22 Oct 2012 14:40:21 +0200
Subject: [PORT FROM MAIN] wifi: Enable Broadcom WiFi solution on Mainline.

BZ: 65640

Enable Broadcom WiFi solution on Mainline respecting
new wifi configuration management.

Change-Id: I1a27aeaa2b66e5fac3bfbecd95e67e98f1c8feef
Orig-Change-Id: Ie205bddb3a6458b63946b3e38af690728b138e2a
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
Reviewed-on: http://android.intel.com:8080/78720
Reviewed-by: Naoumenko, PaulX <paulx.naoumenko@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Jeremie GARCIA <jeremiex.garcia@intel.com>
---
 bcmdhd/Android.mk                      |    3 --
 bcmdhd/bcmdhd_net_iface/Android.mk     |   30 ------------------------
 bcmdhd/config/Android.mk               |   39 --------------------------------
 bcmdhd/dhdutil/Android.mk              |    2 +-
 bcmdhd/firmware/bcm4329/device-bcm.mk  |   29 -----------------------
 bcmdhd/firmware/bcm4330/device-bcm.mk  |   31 -------------------------
 bcmdhd/wpa_supplicant_8_lib/Android.mk |    7 ++---
 7 files changed, 4 insertions(+), 137 deletions(-)
 delete mode 100644 bcmdhd/Android.mk
 delete mode 100644 bcmdhd/bcmdhd_net_iface/Android.mk
 delete mode 100644 bcmdhd/config/Android.mk
 delete mode 100644 bcmdhd/firmware/bcm4329/device-bcm.mk
 delete mode 100644 bcmdhd/firmware/bcm4330/device-bcm.mk

diff --git a/bcmdhd/Android.mk b/bcmdhd/Android.mk
deleted file mode 100644
index 5b43128..0000000
--- a/bcmdhd/Android.mk
+++ /dev/null
@@ -1,3 +0,0 @@
-ifeq ($(BOARD_WLAN_DEVICE),bcmdhd)
-    include $(call all-subdir-makefiles)
-endif
diff --git a/bcmdhd/bcmdhd_net_iface/Android.mk b/bcmdhd/bcmdhd_net_iface/Android.mk
deleted file mode 100644
index be22624..0000000
--- a/bcmdhd/bcmdhd_net_iface/Android.mk
+++ /dev/null
@@ -1,30 +0,0 @@
-#
-# Copyright (C) 2012 Google, inc.
-#
-# $Id: Android.mk,v 2.6 2009-05-07 18:25:15 hharte Exp $
-#
-# Permission to use, copy, modify, and/or distribute this software for any
-# purpose with or without fee is hereby granted, provided that the above
-# copyright notice and this permission notice appear in all copies.
-#
-# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
-# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
-# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
-# SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
-# OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
-# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-#
-
-#
-# lib_net_iface_cmd
-#
-
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-LOCAL_SRC_FILES := bcmdhd_net_iface.c
-LOCAL_MODULE := libnetcmdiface
-LOCAL_CFLAGS := -mabi=aapcs-linux
-LOCAL_MODULE_TAGS := optional
-LOCAL_ALLOW_UNDEFINED_SYMBOLS := true
-include $(BUILD_SHARED_LIBRARY)
diff --git a/bcmdhd/config/Android.mk b/bcmdhd/config/Android.mk
deleted file mode 100644
index b125897..0000000
--- a/bcmdhd/config/Android.mk
+++ /dev/null
@@ -1,39 +0,0 @@
-#
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-LOCAL_PATH := $(call my-dir)
-
-########################
-
-include $(CLEAR_VARS)
-LOCAL_MODULE := dhcpcd.conf
-LOCAL_MODULE_CLASS := ETC
-LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/dhcpcd
-LOCAL_SRC_FILES := android_dhcpcd.conf
-include $(BUILD_PREBUILT)
-
-#########################
-
-WIFI_DRIVER_SOCKET_IFACE := wlan0
-ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_8_X)
-  include external/wpa_supplicant_8/wpa_supplicant/wpa_supplicant_conf.mk
-else
-ifeq ($(strip $(WPA_SUPPLICANT_VERSION)),VER_0_6_X)
-  include external/wpa_supplicant_6/wpa_supplicant/wpa_supplicant_conf.mk
-else
-  include external/wpa_supplicant/wpa_supplicant_conf.mk
-endif
-endif
-#######################
diff --git a/bcmdhd/dhdutil/Android.mk b/bcmdhd/dhdutil/Android.mk
index 16cbc78..9e35d8a 100644
--- a/bcmdhd/dhdutil/Android.mk
+++ b/bcmdhd/dhdutil/Android.mk
@@ -30,6 +30,6 @@ LOCAL_CFLAGS := -DSDTEST -DTARGETENV_android -Dlinux -DLINUX -mabi=aapcs-linux
 LOCAL_C_INCLUDES +=$(LOCAL_PATH)/include
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_OPTIONAL_EXECUTABLES)
-LOCAL_MODULE_TAGS := debug
+LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_EXECUTABLE)
diff --git a/bcmdhd/firmware/bcm4329/device-bcm.mk b/bcmdhd/firmware/bcm4329/device-bcm.mk
deleted file mode 100644
index bce8582..0000000
--- a/bcmdhd/firmware/bcm4329/device-bcm.mk
+++ /dev/null
@@ -1,29 +0,0 @@
-#
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-########################
-
-ifeq ($(strip $(WIFI_BAND)),802_11_ABG)
-BCM_FW_SRC_FILE_STA := fw_bcm4329_abg.bin
-else
-BCM_FW_SRC_FILE_STA := fw_bcm4329.bin
-endif
-BCM_FW_SRC_FILE_AP  := fw_bcm4329_apsta.bin
-
-PRODUCT_COPY_FILES += \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4329/$(BCM_FW_SRC_FILE_STA):system/vendor/firmware/fw_bcmdhd.bin \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4329/$(BCM_FW_SRC_FILE_AP):system/vendor/firmware/fw_bcmdhd_apsta.bin
-########################
diff --git a/bcmdhd/firmware/bcm4330/device-bcm.mk b/bcmdhd/firmware/bcm4330/device-bcm.mk
deleted file mode 100644
index d9acd67..0000000
--- a/bcmdhd/firmware/bcm4330/device-bcm.mk
+++ /dev/null
@@ -1,31 +0,0 @@
-#
-# Copyright (C) 2008 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-########################
--include hardware/broadcom/wlan/bcmdhd/config/config-bcm.mk
-
-ifeq ($(strip $(WIFI_BAND)),802_11_BG)
-BCM_FW_SRC_FILE_STA := fw_bcm4330_bg.bin
-BCM_FW_SRC_FILE_AP  := fw_bcm4330_apsta_bg.bin
-else
-BCM_FW_SRC_FILE_STA := fw_bcm4330_abg.bin
-BCM_FW_SRC_FILE_AP  := fw_bcm4330_apsta_bg.bin
-endif
-
-PRODUCT_COPY_FILES += \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/$(BCM_FW_SRC_FILE_STA):system/vendor/firmware/fw_bcmdhd.bin \
-    hardware/broadcom/wlan/bcmdhd/firmware/bcm4330/$(BCM_FW_SRC_FILE_AP):system/vendor/firmware/fw_bcmdhd_apsta.bin
-########################
diff --git a/bcmdhd/wpa_supplicant_8_lib/Android.mk b/bcmdhd/wpa_supplicant_8_lib/Android.mk
index af3adf9..0583445 100644
--- a/bcmdhd/wpa_supplicant_8_lib/Android.mk
+++ b/bcmdhd/wpa_supplicant_8_lib/Android.mk
@@ -15,8 +15,6 @@
 #
 LOCAL_PATH := $(call my-dir)
 
-ifeq ($(WPA_SUPPLICANT_VERSION),VER_0_8_X)
-
 ifneq ($(BOARD_WPA_SUPPLICANT_DRIVER),)
   CONFIG_DRIVER_$(BOARD_WPA_SUPPLICANT_DRIVER) := y
 endif
@@ -43,8 +41,10 @@ ifdef CONFIG_DRIVER_WEXT
 WPA_SRC_FILE += driver_cmd_wext.c
 endif
 
+ifeq ($(TARGET_ARCH),arm)
 # To force sizeof(enum) = 4
 L_CFLAGS += -mabi=aapcs-linux
+endif
 
 ifdef CONFIG_ANDROID_LOG
 L_CFLAGS += -DCONFIG_ANDROID_LOG
@@ -54,6 +54,7 @@ endif
 
 include $(CLEAR_VARS)
 LOCAL_MODULE := lib_driver_cmd_bcmdhd
+LOCAL_MODULE_TAGS := optional
 LOCAL_SHARED_LIBRARIES := libc libcutils
 LOCAL_CFLAGS := $(L_CFLAGS)
 LOCAL_SRC_FILES := $(WPA_SRC_FILE)
@@ -61,5 +62,3 @@ LOCAL_C_INCLUDES := $(WPA_SUPPL_DIR_INCLUDE)
 include $(BUILD_STATIC_LIBRARY)
 
 ########################
-
-endif
-- 
1.7.4.1

